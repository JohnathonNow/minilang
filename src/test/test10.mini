class: point("point", :X, :Y)

var A := point(10, 20)

print('A:X = {A:X}\n')
print('A = {A}\n')

A:Y := 200

meth (string::of)(P: point) 'point({P:X}, {P:Y})'

print('A = {A}\n')

meth +(A: point, B: point) point(A:X + B:X, A:Y + B:Y)
`
var B := point(23, 9.1)

print('{A} + {B} = {A + B}\n')

meth *(A: point, N: number) point(A:X * N, A:Y * N)
meth *(N: number, A: point) point(N * A:X, N * A:Y)

print('{A} * 5 = {A * 5}\n')
print('7.1 * {B} = {7.1 * B}\n')

class: vector("vector", point, :Z)

var C := vector(1, 2, 3)

print('C = {C}\n')

meth (string::of)(V: vector) 'vector({V:X}, {V:Y}, {V:Z})'

print('C = {C}\n')



