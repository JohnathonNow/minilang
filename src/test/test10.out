A:X = 10
A = point(X: 10, Y: 20)
A = point(10, 200)
point(10, 200) + point(23, 9.1) = point(33, 209.1)
point(10, 200) * 5 = point(50, 1000)
7.1 * point(23, 9.1) = point(163.3, 64.61)
C = point(1, 2)
C = vector(1, 2, 3)
